### DISCUSSION TITLE
Intent 2 analysis

### DISCUSSION THREAD

**Discussion by Kevsy** [camaraproject/EdgeCloud#173] [General]
## Developer intents : Provisioning intents
| Intent 2 | 5SED |5MEE |EXRA|EXRC| EXCA | EXSM|  GMEC |
| -------- |------| ----|----| ---|------|-----|-------|
| 2. "I can discover the capabilities/resources available at a cloudlet/MEC platform: CPU, Memory, Storage, GPU"    | No            |Yes     |Yes           |N/A     |N/A           |N/A     |N/A    |

| 5MEE           | Details       |Comments|
| -------------  |------------| ------|
| `GET /mecplatforms` | Returns a list of optimal MEC Platforms where you can register your deployed application. You can choose to search without passing any of the inputs parameters or a combination of `Service Profile`, `Region`, `subscriber density` or `UEIdentity`.| How can be extracted mec platform resources of a given mec platform? Is it done through service profiles?|

| EXRA           | Details       |Comments|
| -------------  |-------------| ------|
| `POST /api/v1/auth/ctrl/ShowCloudlet` | Show Cloudlets. Lists all the cloudlets managed from Edge Controller.| Response includes Object resource_quotas `{"alert_threshold": 0, "name": "string", "value": 0 }` Would gpu be included here?|

**Comment by ThomasEdgeXR** [comment]
Assumption "MEC" refers to "cloudlets" as defined in https://github.com/camaraproject/EdgeCloud/blob/main/documentation/SupportingDocuments/edge_terminology.md

ShowCloudlet indeed does provide the available flavors available on all cloudlets the developer can access (based on his organization memberships and their cloudlet access rights). The details of the flavors include the required parameters (CPU, Memory, Storage, GPU).

The above comment ("Response includes Object resource..." seems to refer to an execution of ShowCloudlet with Operator Role (i.e. not Developer Role) which returns additional information but is not relevant here.

**Comment by Kevsy** [comment]
Same as for Intent 1 comment,  we can change 'operator MEC' to cloudlet

**Comment by gunjald** [comment]
The resource discovery capability has not been part of the proposed Capgemini APIs i.e. GMEC

**Comment by maheshc01** [comment]
the table accurately reflects the support for the intent in 5GFF MEC Exposure & Experience Management API.

The API supports discovering capabilities/resources available at the MEC site in the context of Application requirements which is defined in the form of service profile.

**Comment by Kevsy** [comment]
updated to say 'Cloudlet/MEC platform' until we have agreed which (or both) to use

**Comment by ThomasEXR** [comment]
This are the flow diagrams for corresponding EdgeXR APIs

showCloudlet (https://api.edgexr.org/#tag/Cloudlet/operation/ShowCloudlet)

```mermaid
sequenceDiagram
    participant developer
    participant operator    
    Note over developer,operator: PRE App determined proper Operator Gateway as per attached network
    Note over developer,operator: PRE App developer registered, <br/> authenticated and authorised
    developer->>operator: POST /ShowCloudlet <region>
    operator->>developer: Iist of cloudlets in the requested region
```
